Mathlib/Algebra/Order/Group/Action.lean

```diff
@@ -0,0 +1,34 @@
+/-
+Copyright (c) 2024 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.Algebra.CovariantAndContravariant
+import Mathlib.Order.ConditionallyCompleteLattice.Basic
+
+/-!
+# Results about `CovariantClass G α HSMul.hSMul LE.le`
+
+When working with group actions rather than modules, we drop the `0 < c` condition.
+
+Notably these are relevant for pointwise actions on set-like objects.
+-/
+
+variable {ι : Sort*} {M α : Type*}
+
+theorem smul_mono_right [SMul M α] [Preorder α] [CovariantClass M α HSMul.hSMul LE.le]
```

Comment:
Can you add a version with the `gcongr` attribute (I think it has to be formulated separately).

bors d+